samples/net/sockets/tls_configurations/sample.yaml

```diff
@@ -0,0 +1,41 @@
+common:
+  tags:
+    - mbedtls
+    - socket
+  platform_allow:
+    - native_sim
+    - native_sim/native/64
+  # Testing requires the OpenSSL to be started, but this is currently not
+  # available on the CI so we skip it. Testing is possible locally though.
+  skip: true
+sample:
+  description: Test Mbed TLS sockets with various TLS versions and key exchanges
+  name: socket-tls-configurations
+tests:
+  sample.net.sockets.tls12.rsa_kex:
+    extra_args:
+      - EXTRA_CONF_FILE=overlay_tls12.conf;overlay_rsa.conf
+    extra_configs:
+      - CONFIG_MBEDTLS_KEY_EXCHANGE_RSA_ENABLED=y
+  sample.net.sockets.tls12.ec_kex:
+    extra_args:
+      - EXTRA_CONF_FILE=overlay_tls12.conf;overlay_ec.conf
+    extra_configs:
+      - CONFIG_MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED=y
+  sample.net.sockets.tls13.ec_kex.no_tickets:
+    extra_args:
+      - EXTRA_CONF_FILE=overlay_tls13.conf;overlay_ec.conf
+    extra_configs:
+      - CONFIG_MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED=y
+  sample.net.sockets.tls13.ec_kex:
+    extra_args:
+      - EXTRA_CONF_FILE=overlay_tls13.conf;overlay_ec.conf
+    extra_configs:
+      - CONFIG_MBEDTLS_SESSION_TICKETS=y
+      - CONFIG_MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED=y
```

Comment:
`CONFIG_MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED` is for TLS 1.2. TLS 1.3 ciphersuite names do not include the key exchange. TLS 1.3 `CONFIG_MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED` allows either ECDH or FFDH key exchanges with ephemeral set (selection of EC or FF depends on which groups are proposed/accepted during the key exchange).
But I agree that I can improve the name of the test as ephemeral because it's better.

Reply:
I also just realized that there was an error in the `no_tickets` case :facepalm: